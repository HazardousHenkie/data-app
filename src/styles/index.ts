import { createGlobalStyle } from 'styled-components'

const GlobalStyle = createGlobalStyle`
  html,
  body,
  #root {
    height: 100%;
  }

  .MuiPopover-root {
    z-index: 1400 !important;
  }
`

export default GlobalStyle
