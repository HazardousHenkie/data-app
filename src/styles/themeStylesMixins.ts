const MuiDrawer = {
    padding: '20px',
    borderTopRightRadius: '25px',
    borderTopLeftRadius: '25px'
}

export default MuiDrawer
