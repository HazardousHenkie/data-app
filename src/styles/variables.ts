const variables = {
    black: '#000',
    white: '#fff',
    gray: '#D3D3D3',
    darkGray: '#A9A9A9',
    breakpoints: {
        up: {
            xxs: 'min-width: 0',
            xs: 'min-width: 420px',
            sm: 'min-width: 600px',
            md: 'min-width: 960px',
            lg: 'min-width: 1280px',
            xl: 'min-width: 1920px'
        },
        down: {
            xs: 'max-width: 420px',
            sm: 'max-width: 599px',
            md: 'max-width: 959px',
            lg: 'max-width: 1279px',
            xl: 'max-width: 1919px'
        }
    }
}

export default variables
