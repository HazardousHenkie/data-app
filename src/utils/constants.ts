enum ENDPOINTS {
    LOGOUT = '/.netlify/functions/googleLogout',
    lOGIN = '/.netlify/functions/googleLogin',
    REFRESH_TOKEN = '/.netlify/functions/refreshToken',
    COUNTRIES = '/.netlify/functions/countries',
    FAVORITE_COUNTRIES = '/.netlify/functions/getFavoritedCountries',
    SAVE_COUNTRY = '/.netlify/functions/saveCountry/',
    DELETE_COUNTRY = '/.netlify/functions/deleteCountry/'
}

export default ENDPOINTS
