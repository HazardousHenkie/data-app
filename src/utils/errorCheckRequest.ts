import { put } from 'redux-saga/effects'
import { getRefreshTokenRequest } from 'globals/authentication/refreshToken/actions'
import { logoutRequest } from 'globals/authentication/logout/actions'

import { setError } from 'globals/globalErrors/actions'
import { ResponseError } from './request'

// test this one?
function* requestErrorCheck(error: ResponseError) {
    try {
        if (localStorage.getItem('userId')) {
            yield put(
                getRefreshTokenRequest(localStorage.getItem('userId') as string)
            )
        } else {
            throw Error('No userId in local storage.')
        }
    } catch {
        yield put(logoutRequest())
        yield put(setError(error))
    }
}
export default requestErrorCheck
