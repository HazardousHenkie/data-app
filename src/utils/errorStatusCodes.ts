enum ERROR_STATUS_CODES {
    BAD_REQUEST = 400,
    UNAUTHORIZED = 401,
    SERVER_ERROR = 500,
    SERVICE_UNAVAILABLE = 503,
    NOT_FOUND = 404
}

export default ERROR_STATUS_CODES
