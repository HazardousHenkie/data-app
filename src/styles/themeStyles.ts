import { createMuiTheme, responsiveFontSizes } from '@material-ui/core/styles'

let lightTheme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      main: '#343434'
    }
  }
})

export let darkTheme = responsiveFontSizes(
  createMuiTheme({
    palette: {
      type: 'dark',
      primary: {
        main: '#343434'
      }
    }
  })
)

export default lightTheme = responsiveFontSizes(lightTheme)
