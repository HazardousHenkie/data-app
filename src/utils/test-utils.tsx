import React from 'react'
import { render } from '@testing-library/react'
import configureStore from 'configureStore'

import history from 'utils/history'
import { Provider } from 'react-redux'

import variables from 'styles/variables'
import { ThemeProvider } from 'styled-components'
import { ThemeProvider as MuiThemeProvider } from '@material-ui/styles'
import lightTheme from 'styles/themeStyles'

const customRender = (
    ui: React.ReactElement,
    {
        initialState = {},
        store = configureStore(initialState, history),
        ...renderOptions
    } = {}
) => {
    const Wrapper: React.FC = ({ children }) => {
        return (
            <Provider store={store}>
                <ThemeProvider theme={variables}>
                    <MuiThemeProvider theme={lightTheme}>
                        <ThemeProvider theme={lightTheme}>
                            {children}
                        </ThemeProvider>
                    </MuiThemeProvider>
                </ThemeProvider>
            </Provider>
        )
    }
    return render(ui, {
        wrapper: Wrapper,
        ...renderOptions
    })
}

export * from '@testing-library/react'

export { customRender as render }
