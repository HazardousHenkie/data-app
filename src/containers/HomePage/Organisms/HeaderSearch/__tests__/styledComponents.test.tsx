import React from 'react'

import { css, ThemeProvider } from 'styled-components'

import { render } from 'utils/test-utils'

import variables from 'styles/variables'

import { Theme } from '@material-ui/core/styles'
import lightTheme, { darkTheme } from 'styles/themeStyles'

import TopBar, {
    CloseCountriesList,
    SearchFieldWrapper,
    RotateNinetyDegrees
} from '../styledComponents'

import 'jest-styled-components'

const renderWithTheme = (theme?: Theme) =>
    render(
        <ThemeProvider theme={theme || lightTheme}>
            <TopBar />
        </ThemeProvider>
    )

const renderCloseCountriesList = (theme?: Theme) =>
    render(<CloseCountriesList theme={theme || lightTheme} />)

describe('<TopBar />', () => {
    it('should render and match the snapshot', () => {
        const { container } = renderWithTheme()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <div> tag', () => {
        const { container } = renderWithTheme()

        expect(container.firstElementChild?.tagName).toEqual('DIV')
    })

    it('should render shadow variable', () => {
        const { container } = renderWithTheme()

        expect(container.firstChild).toHaveStyle(
            `box-shadow: ${variables.shadow}`
        )
    })

    it('should render light variable when LightTheme is set', () => {
        const { container } = renderWithTheme()

        expect(container.firstChild).toHaveStyle(
            `background: ${lightTheme.palette.primary.main}`
        )
    })

    it('should render light variable when darkmode is set', () => {
        const { container } = renderWithTheme(darkTheme)

        expect(container.firstChild).toHaveStyle(
            `background: ${darkTheme.palette.primary.main}`
        )
    })
})

describe('<SearchFieldWrapper />', () => {
    it('should render and match the snapshot', () => {
        const { container } = render(<SearchFieldWrapper />)

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <div> tag', () => {
        const { container } = render(<SearchFieldWrapper />)

        expect(container.firstElementChild?.tagName).toEqual('DIV')
    })
})

describe('<CloseCountriesList />', () => {
    it('should render and match the snapshot', () => {
        const { container } = renderCloseCountriesList()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <SVG> tag', () => {
        const { container } = renderCloseCountriesList()

        expect(container.querySelector('svg')).toBeInTheDocument()
    })

    it('should render light variable when LightTheme is set', () => {
        const { container } = renderCloseCountriesList()

        expect(container.firstChild).toHaveStyle(
            `color: ${lightTheme.palette.common.black}`
        )
    })

    it('should render light variable when darkmode is set', () => {
        const { container } = renderCloseCountriesList(darkTheme)

        expect(container.firstChild).toHaveStyle(
            `color: ${darkTheme.palette.common.black}`
        )
    })
})
