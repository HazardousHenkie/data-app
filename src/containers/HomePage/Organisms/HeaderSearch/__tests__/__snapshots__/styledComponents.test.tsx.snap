// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseCountriesList /> should render and match the snapshot 1`] = `
.c0 {
  font-size: 2.5rem;
  width: 35px;
  margin-right: 7px;
  cursor: pointer;
  -webkit-animation: jizBHZ 0.4s ease-in-out;
  animation: jizBHZ 0.4s ease-in-out;
  color: #000;
}

<svg
  aria-hidden="true"
  class="MuiSvgIcon-root c0"
  focusable="false"
  theme="[object Object]"
  viewBox="0 0 24 24"
>
  <path
    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
  />
</svg>
`;

exports[`<SearchFieldWrapper /> should render and match the snapshot 1`] = `
.c0 {
  width: 100%;
}

<div
  class="c0"
/>
`;

exports[`<TopBar /> should render and match the snapshot 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1350;
  padding: 0 10px;
  margin: 15px;
  width: calc(100% - 30px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 10px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.25);
  background: #fffafa;
}

<div
  class="c0"
/>
`;
